src/main/java/com/tabnine/binary/fetch/BundleDownloader.java

```diff
@@ -23,6 +26,15 @@ public BundleDownloader(TempBundleValidator validator, GeneralDownloader downloa
 
   public Optional<BinaryVersion> downloadAndExtractBundle(String version) {
     String urlString = String.join("/", getBundleServerUrl(), version, TARGET_NAME, "TabNine.zip");
+
+    if (Config.IS_ON_PREM) {
+      if (!StaticConfig.getTabnineEnterpriseHost().isPresent()) {
```

Comment:
you can chain `ifPresen(host ->  return host). orElseget( () -> {
Logger.getInstance(getClass()).warn("Not downloading bundle - on prem version but no host is setup!");
        return Optional.empty();}) `

Reply:
I will move it anyway now